src/Event/Subscriber/OmnipayGatewayRequestSubscriber.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * payment-gateway-logger
+ *
+ * @package    payment-gateway-logger
+ * @author     manzoj
+ * @version    1
+ */
+
+namespace PaymentGatewayLogger\Event\Subscriber;
+
+use Guzzle\Common\Event;
+use PaymentGatewayLogger\Event\Constants;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class OmnipayGatewayRequestSubscriber implements EventSubscriberInterface
+{
+    const PRIORITY = 0;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var string
+     */
+    private $gateway_name;
+
+    /**
+     * OmnipayGatewayRequestSubscriber constructor.
+     *
+     * @param string $gateway_name
+     * @param LoggerInterface $logger
+     */
+    public function __construct($gateway_name, $logger)
+    {
+        $this->logger = $logger;
+        $this->gateway_name = 'omnipay_' . $gateway_name;
+    }
+
+    /**
+     * Returns an array of event names this subscriber wants to listen to.
+     *
+     * The array keys are event names and the value can be:
+     *
+     *  * The method name to call (priority defaults to 0)
+     *  * An array composed of the method name to call and the priority
+     *  * An array of arrays composed of the method names to call and respective
+     *    priorities, or 0 if unset
+     *
+     * For instance:
+     *
+     *  * array('eventName' => 'methodName')
+     *  * array('eventName' => array('methodName', $priority))
+     *  * array('eventName' => array(array('methodName1', $priority), array('methodName2')))
+     *
+     * @return array The event names to listen to
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(
+            Constants::OMNIPAY_REQUEST  => array('onOmnipayRequestBeforeSend', self::PRIORITY),
+            Constants::OMNIPAY_RESPONSE => array('onOmnipayRequestSent', self::PRIORITY),
```

Comment:
the naming of omnipay_**response** => onOmnipay**Request**Sent seems a big confusing...I don't totally understand what exactly is being logged because of that

Reply:
See above comment about renaming `onOmnipayRequestSent`